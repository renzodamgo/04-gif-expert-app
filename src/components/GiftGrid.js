import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem'

export const GiftGrid = ({ category }) => {
    const [images, setImages] = useState([])

    useEffect(() => {
        getGifts()
    }, []) // solo se ejecuta una vez

    const getGifts = async () => {
        const trimCategory = category.replace(/\s/g, '+')
        const url = `https://api.giphy.com/v1/gifs/search?q=${trimCategory}&limit=5&api_key=cHrzYubYZ4nTPGmAmQrGm1aDRtL1Vlr5`
        console.log(url)
        const resp = await fetch(url);
        const { data } = await resp.json();
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })
        console.log(gifs)
        setImages(gifs)
    }

    // getGifts()
    return (
        <>
            <h3> {category}</h3>
            <div className='card-grid'>

                {
                    images.map((img) =>
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    )
                }



            </div>

        </>
    )
}
