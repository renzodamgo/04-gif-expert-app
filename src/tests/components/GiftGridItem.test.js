import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en <GifGridItem/>', () => {
    const title = "Gif de test";
    const url = "https://localhost/algo.jpg"
    test('debe de mostrar el componente correctamente', () => {
        const wrapper = shallow(<GifGridItem title={title} url={url}/>)
        expect(wrapper).toMatchSnapshot();
    })
})