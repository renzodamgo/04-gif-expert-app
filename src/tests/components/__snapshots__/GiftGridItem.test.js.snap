// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-0.5s"
>
  <img
    alt="Gif de test"
    src="https://localhost/algo.jpg"
  />
  <p>
    Gif de test
  </p>
</div>
`;
